File: NumSecreto.java
import java.util.Scanner;
public class NumSecreto {
	public static void main(String[] args) {
		
		Scanner scanner = new Scanner( System.in );

		int secreto;
		int intentos = 0;
		int numPersona = -1;

		secreto = (int)Math.floor(Math.random()* 100)+1;

	
	boolean VolverJugar = true;
	
	while(VolverJugar == true) {
		System.out.println("Ecribe un número del 1 al 100");
		for(intentos = 0; numPersona != secreto ; intentos++){
			
			numPersona = scanner.nextInt();
			if(numPersona <= 0)
				System.out.println("Escribe un valor superior a 0");
			
			if(numPersona > 100)
				System.out.println("Escribe un valor inferior a 100");
			
			if(secreto > numPersona && numPersona > 0)

				System.out.println("¡¡El número es mayor, escribe otro número!!");

	

			if(secreto < numPersona && numPersona < 100)

					System.out.println("¡¡El número es menor!! , escribe otro número");

			if(secreto == numPersona) {
	
		}
		
		}
		System.out.println("¡¡Has Acertado!! , tu cantidad de intentos es " + intentos + " ¿deseas volver a jugar?, si es así, escribe una s y si no desas vovler a jugar escribe una n");
	
		
		String respuesta = scanner.next().toLowerCase();

		if(respuesta.equals("n")) {
			
			VolverJugar = false;
			
	    System.out.println("El juego ha terminado");
		
	    
		}
								
			if(VolverJugar = respuesta.equals("s")) {
				secreto = (int)Math.floor(Math.random()* 100)+1;
				intentos = 0;
				VolverJugar = true;
			}	  		
		
	}	
	}
		
	}

File: numsecreto.rb
def juego_numero_secreto
  secreto = rand(1..100)
  intentos = 0
  num_persona = -1
  volver_jugar = true

  while volver_jugar
    puts "Escribe un número del 1 al 100"
    until num_persona == secreto
      num_persona = gets.to_i
      intentos += 1

      if num_persona <= 0
        puts "Escribe un valor superior a 0"
      elsif num_persona > 100
        puts "Escribe un valor inferior a 100"
      elsif secreto > num_persona && num_persona > 0
        puts "¡¡El número es mayor, escribe otro número!!"
      elsif secreto < num_persona && num_persona < 100
        puts "¡¡El número es menor!!, escribe otro número"
      end
    end

    puts "¡¡Has acertado!! Tu cantidad e intentos es #{intentos}. ¿Deseas volver a jugar? Si es así, escribe una 's', si no, escribe una 'n'."
    respuesta = gets.chomp.downcase

    if respuesta == 'n'
      volver_jugar = false
      puts "El juego ha terminado"
    elsif respuesta == 's'
      secreto = rand(1..100)
      intentos = 0
      num_persona = -1
    end
  end
end

juego_numero_secreto
